mistralrs-core/src/paged_attention/layers/paged_attention.rs

```diff
@@ -70,6 +70,34 @@ impl PagedAttention {
             input_metadata.slot_mappings.clone()
         };
 
+        // When device mapping, these Tensors are fixed on the first device, and must be moved to the same device as q,k,v
```

Comment:
I see you mentioned a performance penalty:
> Currently, device-mapped paged attention is approximately 10% slower compared to single device paged attention. I found the slowdown is at least partially due to the overhead of moving tensors to the device on every layer forward pass.

To avoid this, can you please update `PagedAttentionInputMetadata` to store all tensors as hashmaps of device location to the actual tensor? Do you think this is a good solution?

This takes up more memory on each GPU but requires only one copy (in the inputs processor) and enables us to remove this section. I'm thinking something similar to [this](https://github.com/EricLBuehler/mistral.rs/blob/d28ddf96d9dc79469cd6f1f856ea4d2370819a80/mistralrs-core/src/models/llama.rs#L480-L494) where we create multiple RoPE instantiations on different devices.

Additionally, I just merged #1014. Can you please merge with master to get these new changes, otherwise a conflict will occur with the addition of the changes I requested above.